eda/src/autogluon/eda/analysis/explain.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+import shap
+
+from autogluon.core.constants import REGRESSION
+from autogluon.eda import AnalysisState
+from autogluon.eda.analysis.base import AbstractAnalysis
+
+__all__ = ["ShapAnalysis"]
+
+logger = logging.getLogger(__name__)
+
+
+class _ShapAutogluonWrapper:
+    def __init__(self, predictor, feature_names, target_class=None):
+        self.ag_model = predictor
+        self.feature_names = feature_names
+        self.target_class = target_class
+        if target_class is None and predictor.problem_type != REGRESSION:
+            logging.warning("Since target_class not specified, SHAP will explain predictions for each class")
+
+    def predict_proba(self, X):
+        if isinstance(X, pd.Series):
+            X = X.values.reshape(1, -1)
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X, columns=self.feature_names)
+        if self.ag_model.problem_type == REGRESSION:
+            preds = self.ag_model.predict(X)
+        else:
+            preds = self.ag_model.predict_proba(X)
+        if self.ag_model.problem_type == REGRESSION or self.target_class is None:
+            return preds
+        else:
+            return preds[self.target_class]
+
+
+class ShapAnalysis(AbstractAnalysis):
+    """
+    Perform Shapley values calculation using `shap` package for the given rows.
+
+    Parameters
+    ----------
+    rows: pd.DataFrame,
+        rows to explain
+    baseline_sample: int, default = 100
+        The background dataset size to use for integrating out features. To determine the impact
+        of a feature, that feature is set to "missing" and the change in the model output
```

Comment:
"missing": Do we have a consistent definition for what setting to "missing" means?

Reply:
This is how the `shap` describes the process.